src/app/template/shared/template.service.ts

```diff
@@ -41,4 +54,36 @@ export class TemplateService extends BaseTemplateService {
       Template
     >;
   }
+
+  public getUploadParamsForTemplate(
+    params: TemplateUploadParamsRequest,
+  ): Observable<TemplateUploadParams> {
+    const requestParams = { ...params };
+
+    requestParams['hypervisor'] = requestParams['hypervisor'] || 'KVM';
+    requestParams['format'] = requestParams['format'] || 'QCOW2';
+    requestParams['requiresHvm'] = requestParams['requiresHvm'] || true;
+
+    return this.sendCommand(CSCommands.GetUploadParamsFor, requestParams).pipe(
+      tap(() => this.invalidateCache()),
+      map(({ getuploadparams }) => getuploadparams),
+    ) as Observable<TemplateUploadParams>;
+  }
+
+  public uploadTemplate(params: TemplateUploadParams): Observable<unknown> {
+    const { localTemplate, postURL, expires, metadata, signature } = params;
+
+    const formData = new FormData();
+    formData.append('file', localTemplate);
+
+    const httpClient = new HttpClient(this.httpBackend); // bypass interceptor
```

Comment:
can we at least reprovide it or create a factory provider to avoid manual instantiation?